src/components/v2/Table/index.tsx

```diff
@@ -0,0 +1,72 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState } from 'react';
+import TableMUI from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import Head from './Head';
+import { useStyles } from './styles';
+
+interface ITableRowProps {
+  key: string;
+  render: () => React.ReactNode | string;
+  value: string | number | boolean;
+}
+
+interface ITableProps {
+  title: string;
+  data: ITableRowProps[][];
+  columns: { id: string; label: string; orderable: boolean }[];
+  fetchData?: (params: { order: { by: string; direction: 'asc' | 'desc' } }) => void;
+}
+
+export function Table({ columns, data, fetchData, title }: ITableProps) {
+  const styles = useStyles();
+  const [orderBy, setOrderBy] = useState<typeof columns[number]['id'] | undefined>(undefined);
+  const [order, setOrder] = useState<'asc' | 'desc' | undefined>('asc');
+  const [rows, setRows] = useState(data);
```

Comment:
Instead of storing the ordered rows in a state, you can memoize them and automatically update them when `order`, `orderBy` or the data itself change:

```typescript
const rows = React.useMemo(async () => {
  if (!orderBy) {
    return data;
  }

  const rowIndex = columns.findIndex(column => column.id === orderBy);
  
  return [...data].sort((a, b) => {        
    if (a[rowIndex].value < b[rowIndex].value) {
      return newOrder === 'asc' ? -1 : 1;
    }
    if (a[rowIndex].value > b[rowIndex].value) {
      return newOrder === 'asc' ? 1 : -1;
    }
    return 0;
  });
}, [data, orderBy, order]);
```

Then `onRequestOrder` would only change `orderBy` and `orderDirection`:

```typescript
const onRequestOrder = (property: string) => {
  const newOrder = property === orderBy && order === 'asc' ? 'desc' : 'asc';
  
  setOrderBy(property);
  setOrder(newOrder);
});
```

The reason behind avoiding to keep the ordered rows in a state is that if the `data` prop changes, the table won't reflect the update until the user presses on a header to change the order.